### PR TITLE
Ameba: Miscellaneous fixes

### PR DESCRIPTION
**Pull request by syeopite**

### CODE REVIEW

File: src/invidious/helpers/i18next.cr [hunk 1]
```diff
@@ -189,7 +189,7 @@ module I18next::Plurals
 
     # Emulate the `rule.numbers.size == 2 && rule.numbers[0] == 1` check
     # from original i18next code
-    private def is_simple_plural(form : PluralForms) : Bool
+    private def simple_plural?(form : PluralForms) : Bool
```
Comment (SamantazFox) [inline src/invidious/helpers/i18next.cr:192]: Can you revert that change, as it's already in #4806 ?

File: src/invidious/routes/video_playback.cr [hunk 2]
```diff
@@ -131,7 +131,7 @@ module Invidious::Routes::VideoPlayback
       end
 
       # TODO: Record bytes written so we can restart after a chunk fails
-      while true
+      loop do
```
Comment (SamantazFox) [inline src/invidious/routes/video_playback.cr:134]: Is that change requested by Ameba?
`loop` is just a macro for `while true` anyway:
https://github.com/crystal-lang/crystal/blob/4cea10199d5006000a129413f6f607697185c83a/src/kernel.cr#L114

Comment (syeopite) [reply src/invidious/routes/video_playback.cr:134]: Yep, `Style/WhileTrue`. It does seem like a pretty useless change to me though. 

Should that rule be disabled?

File: src/invidious/helpers/i18next.cr (hunk 1 above)
Comment (syeopite) [reply src/invidious/helpers/i18next.cr]: Done. Sorry about that

File: src/invidious/routes/video_playback.cr (hunk 2 above)
Comment (SamantazFox) [reply src/invidious/routes/video_playback.cr:134]: Nah, in that case it's good to have a tool that unifies the syntax.

File: src/invidious/videos.cr [hunk 3]
```diff
@@ -355,11 +370,12 @@ struct Video
   getset_i64 likes
   getset_i64 views
 
+  # TODO: Make predicate_bool the default as to adhere to Crystal conventions
   getset_bool allowRatings
   getset_bool authorVerified
   getset_bool isFamilyFriendly
   getset_bool isListed
-  getset_bool isUpcoming
+  predicate_bool upcoming, isUpcoming
```
Comment (SamantazFox) [inline src/invidious/videos.cr:378]: Why are you only changing that one? There are multiple other `is*` methods!

Comment (syeopite) [reply src/invidious/videos.cr:378]: This PR is mostly to just fix the Ameba issues which for some reason does not detect the other methods as violating `Naming/PredicateName`

Hence the TODO comment a few lines above

https://github.com/iv-org/invidious/blob/b75e5cfba6b4683307a5755506deefaebee5c3dd/src/invidious/videos.cr#L373

Do you want me to also resolve those in this PR?

Comment (SamantazFox) [reply src/invidious/videos.cr:378]: No, please keep that for a future PR, I'm worried it could cause too much conflicts with other open PRs!

**Review by SamantazFox — APPROVED** [review]

**Comment by syeopite** [comment]
Rebased against master; and all CI has passed. Once merged it looks like Invidious will finally be Ameba compliant!

**Review by SamantazFox — APPROVED** [review]

**Comment by SamantazFox** [comment]
Thanks for your Ameba compliance PR series!

**Pull request merged**